src/hyperlight_host/src/mem/shared_mem.rs

```diff
@@ -437,10 +438,24 @@ impl ExclusiveSharedMemory {
         };
 
         #[cfg(inprocess)]
-        let addr =
-            unsafe { MapViewOfFile(handle, FILE_MAP_ALL_ACCESS | FILE_MAP_EXECUTE, 0, 0, 0) };
+        let handle = unsafe {
+            CreateFileMappingA(
```

Comment:
nit: Seems like this CreateFileMappingA and the below MapViewOfFile are the same inprocess and when running out of process, except for a flag. If you want, you could just change the flag depending on the #cfg instead of the entire funcitno call.

Reply:
Fixed and added a missing error check